typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts

```diff
@@ -53,13 +54,57 @@ enum ViewSide {
     Top,
 }
 
+function MakeExpLabel(a: number): string {
+    let label = d3.format(".2e")(a);
+    label = label.replace("e", "E");
+    label = label.replace("\u2212", "-");
+    return label;
+}
+
+function UseExponential(
+    a: number,
+    b: number,
+    formatLabelFunc: (x: number) => string | undefined
+): [boolean, boolean, number] {
+    let log10 = Math.log(Math.abs(a)) / Math.log(10); // log base 10.
+    const noDigitsNeededA = Math.ceil(log10) + (a < 0 ? 1 : 0); // add a digit for '-' char.
+
+    log10 = Math.log(Math.abs(b)) / Math.log(10);
+    const noDigitsNeededB = Math.ceil(log10) + (b < 0 ? 1 : 0);
+
+    const noDigitsNeeded =
+        noDigitsNeededA >= noDigitsNeededB ? noDigitsNeededA : noDigitsNeededB;
+    const n = noDigitsNeededA >= noDigitsNeededB ? a : b;
+
+    const eLabel = MakeExpLabel(n);
+
+    const expLength = eLabel.length;
+
+    const useExponential = expLength < noDigitsNeeded;
+    const useFracDigits = noDigitsNeeded < 5;
+
+    let maxLength = useExponential
+        ? expLength
+        : noDigitsNeeded + (useFracDigits ? 1 : 0);
+
+    if (formatLabelFunc) {
+        const label = formatLabelFunc(n);
+        maxLength = label?.length ?? 0;
+    }
+
+    return [useExponential, useFracDigits, maxLength];
+}
+
 const zDepthAxesBackground = 90;
 const zDepthAxesAxis = zDepthAxesBackground + 1;
 const zDepthAxesTicksAndLabels = zDepthAxesAxis + 1;
+const tickLineLength = 10;
+const pixelScale = 8;
 
 export interface Axes2DLayerProps extends ExtendedLayerProps {
-    marginH: number;
-    marginV: number;
+    marginH: number | undefined;
+    marginV: number | undefined;
+    formatLabelFunc: (x: number) => string | undefined;
```

Comment:
Good that we have the ability to override. As long as we have that, maybe we don't need to change the default?

Reply:
I think the option to not define the margins is nice so that the margins get auto calculated? 
That happen if one uses the formatLabelfunc or not.

Reply:
I agree that the automatic margins are useful. I just wonder if we need to implement a new default label formatting behaviour if we can instead stick with the current default behaviour and override it instead? You can resolve.